packages/repository/src/__tests__/unit/repositories/relations-helpers/mongo-related-helpers.unit.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository-tests
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {ObjectID} from 'bson';
+import {belongsTo, Entity, hasMany, model, property} from '../../../..';
+import {
+  isBsonType,
+  deduplicate,
+  normalizeKey,
+  buildLookupMap,
+  reduceAsArray,
+} from '../../../../relations';
+
+describe('unit tests, simulates mongodb env for helpers of ininclusion resolver ', () => {
+  describe('helpers for formating instances', async () => {
+    it('isBsonType', async () => {
+      const id1 = new ObjectID();
+      const id2 = new ObjectID();
+      expect(isBsonType(id1)).to.be.true();
+      expect(isBsonType(id2)).to.be.true();
+    });
+    it('simple deduplcate + isBsonType', async () => {
+      const id1 = new ObjectID();
+      const id2 = new ObjectID();
+
+      const result = deduplicate([id1, id2]);
+
+      expect(result).to.deepEqual([id1, id2]);
+    });
+    it('multiple items deduplcate + isBsonType', async () => {
+      const id1 = new ObjectID();
+      const id2 = new ObjectID();
+      const id3 = new ObjectID();
+      expect(isBsonType(id1)).to.be.true();
+
+      const result = deduplicate([id3, id1, id1, id3, id2]);
+
+      expect(result).to.deepEqual([id3, id1, id2]);
+    });
+  });
+
+  describe('helpers for generating inclusion resolvers', async () => {
+    it('normalizeKey', async () => {
+      const id = new ObjectID();
+      expect(normalizeKey(id)).to.eql(id.toHexString());
+    });
+    // the tests below simulate mongodb environment.
+    it('simple normalizeKey + buildLookupMap', async () => {
+      const id = new ObjectID();
+      expect(normalizeKey(id)).to.eql(id.toHexString());
```

Comment:
This test looks like a duplicate of the previous test on lines 45-49. What am I missing? Can we keep just one of them please?